### PR TITLE
chore: fixed high contrast on web components

### PR DESCRIPTION
**Pull request by khamudom**
address high contrast issues for the calendar, checkbox, data-grid-row, menu-item, and switch

## Related Issue(s)

Fixes #

### CODE REVIEW

**Review by EisenbergEffect — APPROVED** [review]

File: packages/web-components/src/menu-item/menu-item.styles.ts [hunk 1]
```diff
@@ -251,24 +251,25 @@ export const menuItemStyles: (context: ElementDefinitionContext, definition: Men
...
+        :host(:hover) .checkbox,
+        :host(:hover) .radio,
+        :host(:${focusVisible}) .expanded-toggle,
+        :host(:${focusVisible}) .checkbox,
+        :host(:${focusVisible}) .radio,
+        :host([checked]:hover) .checkbox,
+        :host([checked]:hover) .radio,
+        :host([checked]:${focusVisible}) .checkbox,
+        :host([checked]:${focusVisible}) .radio {
           border-color: ${SystemColors.HighlightText};
```
Comment (eljefe223) [inline packages/web-components/src/menu-item/menu-item.styles.ts:263]: Should this be using outline?

Comment (eljefe223) [reply packages/web-components/src/menu-item/menu-item.styles.ts:263]: @chrisdholt

**Review by eljefe223 — APPROVED** [review]

File: packages/web-components/src/menu-item/menu-item.styles.ts (hunk 1 above)
Comment (khamudom) [reply packages/web-components/src/menu-item/menu-item.styles.ts:263]: I will actually remove this and other selectors that are styling this to look like a checkbox. 
It should not look like a checkbox. It should just be an indicator that the menu item is checked. 
Here is the screenshot of the change.
image

File: packages/web-components/src/calendar/calendar.styles.ts [hunk 2]
```diff
@@ -148,14 +150,59 @@ ${display("inline-block")} :host {
...
+            background: ${SystemColors.ButtonFace};
+            color: ${SystemColors.ButtonText};
+            outline: 1px solid ${SystemColors.Highlight};
+        }
+        .week-day:${focusVisible},
+        .interact .day:not(.disabled, .inactive):${focusVisible} {
+            forced-color-adjust: none;
+            background: :${SystemColors.ButtonFace};
+            border-color: transparent;
+            box-shadow: 0 0 0 calc(${focusStrokeWidth} * 1px) ${SystemColors.Highlight} inset;
```
Comment (bheston) [inline packages/web-components/src/calendar/calendar.styles.ts:175]: Let's look again at the use of `outline` and `box-shadow` for focus state. With one of my recent PRs the focus state should be through two helpers now. For this component there's space to use the normal treatment `focusTreatmentBase`. Just set `outline-color` in HC styles.

File: packages/web-components/src/calendar/calendar.styles.ts [hunk 3]
```diff
@@ -148,14 +150,59 @@ ${display("inline-block")} :host {
...
-          }
-
-          .today .date {
-              background: ${SystemColors.Highlight};
-              color: ${SystemColors.HighlightText};
-          }
+        :host {
+          forced-color-adjust: auto;
+          background: ${SystemColors.ButtonFace};
+          outline: 1px solid ${SystemColors.CanvasText};
```
Comment (bheston) [inline packages/web-components/src/calendar/calendar.styles.ts:156]: Is this adding a border similar to a card? If so, I think we want to have a transparent border in the base styles and then set the color here. Let's try to keep `outline` for focus state only.

Comment (khamudom) [reply packages/web-components/src/calendar/calendar.styles.ts:156]: updated and set a transparent border in default contrast

File: packages/web-components/src/calendar/calendar.styles.ts (hunk 2 above)
Comment (khamudom) [reply packages/web-components/src/calendar/calendar.styles.ts:175]: removed box-shadow and outline and used `focusTreatmentBase`

File: packages/web-components/src/calendar/calendar.styles.ts [hunk 4]
```diff
@@ -148,14 +151,59 @@ ${display("inline-block")} :host {
...
+        .week-day {
+            background: ${SystemColors.Canvas};
+            color: ${SystemColors.CanvasText};
+            fill: currentcolor;
+            border-color: ${SystemColors.Canvas}
+        }
+        .interact .day:not(.disabled, .inactive):hover {
+            background: ${SystemColors.ButtonFace};
+            color: ${SystemColors.ButtonText};
+            outline: 1px solid ${SystemColors.Highlight};
```
Comment (bheston) [inline packages/web-components/src/calendar/calendar.styles.ts:169]: In the latest pattern for the styles, `outline` should only be used for focus treatment. This doesn't appear to be a focused state, so it should probably use `border` instead. Is the border needed for the regular styles or only for HC?

File: packages/web-components/src/calendar/calendar.styles.ts [hunk 5]
```diff
@@ -148,14 +151,59 @@ ${display("inline-block")} :host {
...
+        }
+        .week-day:${focusVisible},
+        .interact .day:not(.disabled, .inactive):${focusVisible} {
+            forced-color-adjust: none;
+            background: ${SystemColors.ButtonFace};
+            border-color: transparent;
+            ${focusTreatmentBase}
```
Comment (bheston) [inline packages/web-components/src/calendar/calendar.styles.ts:176]: If an element is able to be focused, this helper should be applied on the regular styles only. For HC, only set `outline-color` to the necessary `SystemColors` value.

File: packages/web-components/src/calendar/calendar.styles.ts [hunk 6]
```diff
@@ -148,14 +151,59 @@ ${display("inline-block")} :host {
...
+            outline-color: ${SystemColors.Highlight}
+        }
+        .day.disabled,
+        .day.inactive {
+            color: ${SystemColors.GrayText};
+            opacity:1;
+            outline: none;
+        }
+        .day.inactive:${focusVisible} {
+            outline: calc(${focusStrokeWidth} * 1px) solid ${SystemColors.GrayText};
```
Comment (bheston) [inline packages/web-components/src/calendar/calendar.styles.ts:186]: Same comment as above. It's possible the regular styles need some cleanup here. There should be no `box-shadow` to remove either (if the regular styles are up to date).

File: packages/web-components/src/calendar/calendar.styles.ts [hunk 7]
```diff
@@ -148,14 +151,59 @@ ${display("inline-block")} :host {
...
+            box-shadow: none;
+        }
+        .day.selected,
+        .day.selected:not(.disabled, .inactive):hover {
+            color: ${SystemColors.Highlight};
+            outline: 1px solid ${SystemColors.Highlight}
```
Comment (bheston) [inline packages/web-components/src/calendar/calendar.styles.ts:192]: Same. Avoid `outline` for anything other than focus state.

File: packages/web-components/src/calendar/calendar.styles.ts [hunk 8]
```diff
@@ -148,14 +151,59 @@ ${display("inline-block")} :host {
...
+        :host {
+          forced-color-adjust: auto;
+          background: ${SystemColors.ButtonFace};
+        }
+        .day,
+        .interact .day,
+        .week-day {
+            background: ${SystemColors.Canvas};
+            color: ${SystemColors.CanvasText};
+            fill: currentcolor;
```
Comment (bheston) [inline packages/web-components/src/calendar/calendar.styles.ts:163]: This is probably not needed. It should come from the regular styles.

File: packages/web-components/src/calendar/calendar.styles.ts [hunk 9]
```diff
@@ -148,14 +151,59 @@ ${display("inline-block")} :host {
...
+            outline: 1px solid ${SystemColors.Highlight}
+        }
+        .interact .today,
+        .today,
+        .today .date ,
+        .interact .today:not(.disabled, .inactive):hover {
+            forced-color-adjust: none;
+            background: ${SystemColors.Highlight};
+            color: ${SystemColors.HighlightText};
+            fill: currentcolor;
```
Comment (bheston) [inline packages/web-components/src/calendar/calendar.styles.ts:201]: Same as above.

File: packages/web-components/src/calendar/calendar.styles.ts [hunk 10]
```diff
@@ -148,14 +151,59 @@ ${display("inline-block")} :host {
...
+        }
+        .interact .today:not(.disabled, .inactive):${focusVisible} {
+            box-shadow: 0 0 0 calc((${focusStrokeWidth} - ${strokeWidth}) * 1px) ${SystemColors.Highlight} inset,
```
Comment (bheston) [inline packages/web-components/src/calendar/calendar.styles.ts:204]: Same as above.

File: packages/web-components/src/switch/switch.styles.ts [hunk 11]
```diff
@@ -224,21 +224,26 @@ export const switchStyles: (context: ElementDefinitionContext, definition: Switc
...
         }
-        :host(.disabled) slot[name='switch'] {
+        :host(.disabled) slot[name='switch'],
+        :host(.disabled) .label,
+        :host(.disabled) ::slotted([slot="checked-message"]),
+        :host(.disabled) ::slotted([slot="unchecked-message"]) {
           forced-color-adjust: none;
-          fill: ${SystemColors.GrayText};
+          color: ${SystemColors.GrayText};
+          fill: currentcolor;
```
Comment (bheston) [inline packages/web-components/src/switch/switch.styles.ts:246]: Check if this is needed. Should come from regular styles.

**Review by bheston — CHANGES_REQUESTED** [review]

File: packages/web-components/src/calendar/calendar.styles.ts (hunk 7 above)
Comment (khamudom) [reply packages/web-components/src/calendar/calendar.styles.ts:192]: switched to border-color

File: packages/web-components/src/calendar/calendar.styles.ts (hunk 8 above)
Comment (khamudom) [reply packages/web-components/src/calendar/calendar.styles.ts:163]: only have a `.day` selector and only set the `border-color`

File: packages/web-components/src/calendar/calendar.styles.ts (hunk 9 above)
Comment (khamudom) [reply packages/web-components/src/calendar/calendar.styles.ts:201]: this is still needed for this style
image

 if we remove this, then we will not see any item that is considered "today" will not looked selected.

File: packages/web-components/src/calendar/calendar.styles.ts (hunk 5 above)
Comment (khamudom) [reply packages/web-components/src/calendar/calendar.styles.ts:176]: created a focus visible selector in regular styles and removed the helper here.

File: packages/web-components/src/calendar/calendar.styles.ts (hunk 4 above)
Comment (khamudom) [reply packages/web-components/src/calendar/calendar.styles.ts:169]: switch to border-color

File: packages/web-components/src/calendar/calendar.styles.ts (hunk 6 above)
Comment (khamudom) [reply packages/web-components/src/calendar/calendar.styles.ts:186]: removed the selector

File: packages/web-components/src/calendar/calendar.styles.ts (hunk 10 above)
Comment (khamudom) [reply packages/web-components/src/calendar/calendar.styles.ts:204]: removed the selector

File: packages/web-components/src/switch/switch.styles.ts (hunk 11 above)
Comment (khamudom) [reply packages/web-components/src/switch/switch.styles.ts:246]: This is still needed

File: tools/generators/migrate-converged-pkg/index.spec.ts [hunk 12]
```diff
@@ -972,15 +972,15 @@ describe('migrate-converged-pkg generator', () => {
         pkgJson = getPackageJson();
 
         expect(pkgJson.exports).toMatchInlineSnapshot(`
-                  Object {
```
Comment (Hotell) [inline tools/generators/migrate-converged-pkg/index.spec.ts:975]: not sure how this happened but should be reverted

**Review by Hotell — CHANGES_REQUESTED** [review]

**Pull request closed without merging**